code/game/objects/items.dm

```diff
@@ -760,6 +761,7 @@
 		give_item_action(action, user, slot)
 
 	item_flags |= IN_INVENTORY
+	RegisterSignals(src, list(SIGNAL_ADDTRAIT(TRAIT_EXAMINE_SKIP), SIGNAL_REMOVETRAIT(TRAIT_EXAMINE_SKIP)), PROC_REF(update_slot_icon), override = TRUE)
```

Comment:
You used the wrong trait, you put in `TRAIT_EXAMINE_SKIP` and not `TRAIT_NO_WORN_ICON`

Reply:
guh 😭